docs/modules/ROOT/pages/reference/arch-control-plane.adoc

```diff
@@ -0,0 +1,399 @@
+= Central Control Plane (CCP)
+
+This page discusses the architecture of the Central Control Plane (CCP).
+
+* TODO: Where to store consensus to GTC?
+
+image::universal-control-plane.drawio.svg[]
+
+== Summary
+
+The Central Control Plane (CCP) provides the multi-tenant multi-service-provider aggregation and central entry-point.
+All access to the Meta VSHN Application Marketplace happens through this control plane.
+It offers features needed to separate organizations, control access, service provisioning and gives a central view over multi cloud providers and the instances running this way.
+
+We inherit the original architecture of the https://kb.vshn.ch/appuio-cloud/references/architecture/control-api.html[APPUiO Control API^] and extend it for the CCP.
```

Comment:
Is the goal that the CCP replaces the APPUiO Control API, or will these two APIs exist in parallel?

if the goal is to have the CPP replace the APPUiO Control API, there's a bunch of open questions: e.g. how to restrict organizations which are created via e.g. Exoscale Marketplace from creating namespaces on the APPUiO zones? How to restrict APPUiO organizations from creating services on the CSP worker clusters?

Reply:
> Is the goal that the CCP replaces the APPUiO Control API, or will these two APIs exist in parallel?

It will replace the APPUiO Control API. Or in other words: It will advance the APPUiO Control API into a "VSHN Multi-CSP Self-Service API". In essence, APPUiO will become a product in the catalog "Namespace as a Service by VSHN aka APPUiO".

You're right, this part is missing in the docs, I need to dig a bit into what this means for APPUiO specifics today.

Reply:
> In essence, APPUiO will become a product in the catalog "Namespace as a Service by VSHN aka APPUiO".

It's not clear to me that we can simply make APPUiO namespaces ("Namespace as a Service by VSHN") a product that functions the same as other marketplace services, since currently the base design of APPUiO is implemented as "make it as close to vanilla K8s as possible" and allows users to create arbitrary namespaces directly on the APPUiO zones without interacting with the control-api at all.

Reply:
I push this part to be detailed out a bit into the near future. For now, I focus on the Marketplace part. How we proceed (technically) with the APPUiO case will be handled in a follow-up research.

Currently, I think this will end-up similar to what you wrote: It will be handled as selective access to APPUiO clusters (zones). When access is granted, it continues to function as today.